echopype/echodata/zarr_combine.py

```diff
@@ -0,0 +1,742 @@
+from collections import defaultdict
+from typing import Dict, Hashable, List, Optional, Set, Tuple
+
+import dask
+import dask.array
+import numpy as np
+import pandas as pd
+import xarray as xr
+import zarr
+
+from ..utils.coding import COMPRESSION_SETTINGS
+from ..utils.io import get_zarr_compression
+from ..utils.prov import echopype_prov_attrs
+from .api import open_converted
+from .echodata import EchoData
+
+
+class ZarrCombine:
+    """
+    A class that combines a list of EchoData objects by
+    creating a Zarr store and appending each group's
+    Dataset to the store.
+    """
+
+    def __init__(self):
+
+        # all possible time dimensions
+        self.possible_time_dims = {"time1", "time2", "time3", "ping_time"}
+
+        # all possible dimensions that we will append to (mainly time dims)
+        self.append_dims = {"filenames"}.union(self.possible_time_dims)
+
+        # encodings associated with lazy loaded variables
+        self.lazy_encodings = ["chunks", "preferred_chunks"]
+
+        # defaultdict that holds every group's attributes
+        self.group_attrs = defaultdict(list)
+
+        # The sonar_model for the new combined EchoData object
+        self.sonar_model = None
+
+    def _check_ascending_ds_times(self, ds_list: List[xr.Dataset], ed_name: str) -> None:
```

Comment:
Thinking to avoid confusing it'd be good to call this `_check_ascending_ds_sequence` or something similar so that it is not confused with `orchestrate_reverse_time_check` that actually checked all the timestamps.

Reply:
In my mind reverse and ascending are two different properties. For example, the times can be ascending amongst the datasets, but still have reversed times in them. 

`_check_ascending_ds_sequence` is a little to general in my opinion, since we are specifically checking only the times.

I cannot think of a better name, but I am open to other suggestions.

Reply:
Sure I am fine with the current time.